samples/javascript_nodejs/57.teams-conversation-bot/bots/teamsConversationBot.js

```diff
@@ -33,13 +34,65 @@ class TeamsConversationBot extends TeamsActivityHandler {
         });
 
         this.onMembersAddedActivity(async (context, next) => {
+            //  console.log(context);
             context.activity.membersAdded.forEach(async (teamMember) => {
                 if (teamMember.id !== context.activity.recipient.id) {
-                    await context.sendActivity(`Welcome to the team ${ teamMember.givenName } ${ teamMember.surname }`);
+                    await context.sendActivity(
+                        `Welcome to the team ${teamMember.givenName} ${teamMember.surname}`
+                    );
                 }
             });
             await next();
         });
+
+        //Subscribe to different conversation events
+
+        this.onTeamsChannelCreatedEvent(async (context, next) => {
+            await this.onTeamsChannelCreated(context);
+            await next();
+        });
+
+        this.onTeamsChannelRenamedEvent(async (context, next) => {
+            await this.onTeamsChannelRenamed(context);
+            await next();
+        });
+        this.onTeamsChannelDeletedEvent(async (context, next) => {
+            await this.onTeamsChannelDeleted(context);
+            await next();
+        });
+        this.onTeamsChannelRestoredEvent(async (context, next) => {
+            await this.onTeamsChannelRestored(context);
+            await next();
+        });
+
+        this.onTeamsTeamRenamedEvent(async (context, next) => {
+            await this.onTeamsTeamRenamed(context);
+            await next();
+        });
```

Comment:
I don't *believe* these are necessary since [`dispatchConversationUpdateActivity()`](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/teamsActivityHandler.ts#L447) should forward those events to the overrides you have below.

Reply:
Do you want me to implement dispathConversationUpdateActitvity() in this sample as I've tested this

Reply:
`dispathConversationUpdateActitvity()` is already a part of TeamsActivityHandler. I *believe* you can remove all of the `this.onTeams...()` methods *without* needing to add in `dispathConversationUpdateActitvity()`.

Reply:
This is resolved now.